src/Featurizers/TruncatedSVDFeaturizer.h

```diff
@@ -0,0 +1,382 @@
+    MatrixT const                                    _singularvalues;
+    MatrixT const                                    _singularvectors;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+     // MSVC has problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        if (input.size() == 0)
+            throw std::invalid_argument("Input matrix size() invalid");
+
+        if (input.cols() != _singularvectors.rows())
+            throw std::invalid_argument("Input matrix cols() invalid");
+
+        callback(input * _singularvectors);
+    }
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         TruncatedSVDEstimator
+///  \brief         Estimator that reads an annotation created by the `SVDComponentsEstimator`
+///                 and creates a `SVDTransformer` object.
+///
+template <
+    typename MatrixT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class TruncatedSVDEstimator : public TransformerEstimator<MatrixT, MatrixT> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType                          = TransformerEstimator<MatrixT, MatrixT>;
+    using TransformerType                   = TruncatedSVDTransformer<MatrixT>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    TruncatedSVDEstimator(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex);
+    ~TruncatedSVDEstimator(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(TruncatedSVDEstimator);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    size_t const                            _colIndex;
+    MatrixT                                 _matrix;
+    MatrixT                                 _singularValues;
+    MatrixT                                 _singularVectors;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+    bool begin_training_impl(void) override;
+    // MSVC has problems when the declaration and definition are separated
+    FitResult fit_impl(typename BaseType::InputType const *pBuffer, size_t) override {
+        _matrix = *pBuffer;
```

Comment:
>_matrix = *pBuffer; [](start = 8, length = 19)

This line concerns me. If training really only supports 1 input, then we should have code that throws an exception if multiple are found. However, let's deep dive on this to ensure that the desired behavior is correctly implemented.